### PR TITLE
Allow configuring credentials for gradle plugin repository

### PR DESCRIPTION
**Pull request by welandaz**
It is now possible to specify Gradle Plugin repository credentials via Bamboo's Shared Credentials.
This PR also updates the selector to choose the Shared Credential Name for Develocity Access Key.

Screenshot 2024-04-01 at 14 48 03

### CODE REVIEW

File: src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java [hunk 1]
```diff
@@ -123,13 +128,30 @@ private void inject(BuildContext buildContext, Collection<RuntimeTaskDefinition>
...
+                it -> {
+                    UsernameAndPassword credentials = credentialsProvider.findByName(it).orElse(null);
+                    if (credentials == null) {
+                        LOGGER.warn("Plugin repository credentials with the name {} are not found.", it);
+                    } else {
+                        if (credentials.getUsername() == null || credentials.getPassword() == null) {
+                            LOGGER.warn("Plugin repository credentials {} do not have username or password set.", it);
+                        } else {
+                            variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_USERNAME", credentials.getUsername());
+                            variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_PASSWORD", credentials.getPassword());
```
Comment (alextu) [inline src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java:151]: Do we know if this `VariableContext` is shown somewhere in the UI or logs? If so, will the password be masked?

File: src/main/resources/templates/views/admin/buildScansConfig.ftl [hunk 2]
```diff
@@ -31,6 +31,7 @@
         [@ww.textfield labelKey="develocity.config.develocity-plugin.version" name="develocityPluginVersion"/]
         [@ww.textfield labelKey="develocity.config.ccud-plugin.version" name="ccudPluginVersion"/]
         [@ww.textfield labelKey="develocity.config.plugin-repository" name="pluginRepository"/]
+        [@ww.textfield labelKey="develocity.config.plugin-repository-credential-name" name="pluginRepositoryCredentialName"/]
```
Comment (alextu) [inline src/main/resources/templates/views/admin/buildScansConfig.ftl:34]: Is it possible to instead have a dropdown list with the list of available credentials?

Comment (welandaz) [reply src/main/resources/templates/views/admin/buildScansConfig.ftl:34]: It's implemented now as we have it for the Access Key, but your question is still valid.
I'll play around with it and see if we can query the `CredentialsAccessor` and list them in UI somehow.

File: src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java (hunk 1 above)
Comment (welandaz) [reply src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java:151]: Hm, I'll see if I can find them somewhere.

File: src/main/resources/templates/views/admin/buildScansConfig.ftl (hunk 2 above)
Comment (welandaz) [reply src/main/resources/templates/views/admin/buildScansConfig.ftl]: It works.
I've introduced the default "None" value to the list, which sends empty text.
I also changed the "Shared Credential Name" field to be the same selector.

Comment (alextu) [reply src/main/resources/templates/views/admin/buildScansConfig.ftl]: Nice!

**Review by alextu — APPROVED** [review]

File: src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java (hunk 1 above)
Comment (welandaz) [reply src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java:151]: Didn't find it anywhere in the UI or logs of the job (our init script logic doesn't log it)

**Pull request merged**